src/journal/JournalMetadata.cc

```diff
@@ -1022,6 +1030,79 @@ void JournalMetadata::handle_notified(int r) {
   ldout(m_cct, 10) << "notified journal header update: r=" << r << dendl;
 }
 
+void JournalMetadata::disconnect_laggy_clients() {
+  assert(m_lock.is_locked());
+
+  ldout(m_cct, 20) << __func__ << dendl;
+
+  if (m_client_object_sets_behind_max <= 0) {
+    return;
+  }
+
+  std::queue<std::string> clients;
+  for (auto &c : m_registered_clients) {
+    if (c.id == m_client_id ||
```

Comment:
I think we need a configurable filter to exclude certain peers from being disconnected.  We would never want to disconnect the ```librbd::Journal::IMAGE_CLIENT_ID```, only other clients.